src/main/java/com/dsec/backend/service/JobServiceImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.dsec.backend.service;
+
+import com.dsec.backend.entity.Job;
+import com.dsec.backend.entity.Repo;
+import com.dsec.backend.entity.UserEntity;
+import com.dsec.backend.exception.EntityMissingException;
+import com.dsec.backend.exception.ForbidenAccessException;
+import com.dsec.backend.exception.UnauthorizedAccessException;
+import com.dsec.backend.repository.JobRepository;
+import com.dsec.backend.repository.RepoRepository;
+import com.dsec.backend.security.UserPrincipal;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Slf4j
+public class JobServiceImpl implements JobService {
+
+    private final JobRepository jobRepo;
+
+    private final RepoRepository repoRepo;
+
+    @Override
+    public List<Job> getJobsByRepoID(long repoID, Jwt jwt) {
+
+        Repo repo = repoRepo.findById(repoID)
+                .orElseThrow(() -> new EntityMissingException(Repo.class, repoID));
+
+
+        if(!isUserInRepo(repo, UserPrincipal.fromClaims(jwt.getClaims()).getUserEntity()))
+            throw new UnauthorizedAccessException();
```

Comment:
This should be a ForbiddenAccessException. It has a different meaning then anauthorized and it has a handler that returns 403 FORBIDDEN, not 401 UNAUTHORIZED. The difference is in the meaning of the code. 401 = bad credentials, 403 = good credentials, access denied. https://auth0.com/blog/forbidden-unauthorized-http-status-codes/